### PR TITLE
Impl Maven

### PR DESCRIPTION
**Pull request by DeJayDev**
When I first saw this project it made me itch to my very core.

I've implemented Maven for dependency handling rather than the need for.... the weird Downloader thing. I'll be completely ripping that in a following commit.

I want to transform this to a project that better follows Java conventions. Any standard Java IDE can import the project anew by selecting the `pom.xml` rather than the project's root directory. I hope that the following commits help transform this into an amazing twitchdev project.

### CODE REVIEW

**Comment by Dkamps18** [comment]
Thanks, yea i write stuff my own weird way i appreciate taking the time to fix it!

**Comment by Dkamps18** [comment]
The plugin.yml is a requirement otherwise the plugin can't be loaded. also what's the problem with capitalization?

**Comment by DeJayDev** [comment]
The plugin.yml is safe in the resources folder as per this snippet in pom.xml
```xml
    <resources>
      <resource>
        <directory>src/main/resources</directory>
        <filtering>true</filtering>
      </resource>
    </resources>
```

Though nothing has been tested (re: draft), I'm sure it's fine.

Naturally there's no REAL problem with the capitalization aside from [following convention](https://docs.oracle.com/javase/tutorial/java/package/namingpkgs.html).

**Comment by DeJayDev** [comment]
I think I'm all done, let me know if there's any questions or something you want me to implement/fix. 

There are some things I think could use some attention (for example: using a better HTTP library) but I don't want to disrespect the project and uproot everything.

**Comment by Dkamps18** [comment]
Alright i prefer the capitalized version just for the consistency with other projects and how i'd write it generally apart from improving readability which really helps me properly read it which often fails with the non capitalized version due to being dyslectic.

**Comment by DeJayDev** [comment]
Respect, I'll undo that specifically.

**Comment by DeJayDev** [comment]
I have no clue what these conflicts are because I can't write here

**Comment by Dkamps18** [comment]
src/net/Dkamps18/ChannelPoints/Config.java
src/net/Dkamps18/ChannelPoints/Handler/PubSubHandler.java
src/net/Dkamps18/ChannelPoints/Util/DBUtil.java
src/net/Dkamps18/ChannelPoints/main.java

probably these 2 commits
https://github.com/Dkamps18/MC-ChannelPoints/commit/46d7987651b30f16e9510da6786f3b864bf357dc
https://github.com/Dkamps18/MC-ChannelPoints/commit/f51e3c9503a72df754acd196621d30a97525a21b

**Comment by DeJayDev** [comment]
WOW now I'm done.

File: src/main/java/net/dkamps18/channelpoints/Handler/CommandHandler.java [hunk 1]
```diff
@@ -39,13 +37,13 @@ public boolean run(CommandSender sender, Command cmd, String label, String[] arg
...
 			case "connect": {
-				ResultSet res = this.plugin.dbu.get_user_data(p.getUniqueId().toString());
+				ResultSet res = this.plugin.databaseUtil.getUserData(p.getUniqueId().toString());
 				try {
 					if (res.next()) {
-						this.plugin.PubSub.addStream(p, res.getString("twitchid"), res.getString("authtoken"));
+						this.plugin.pubSubHandler.addStream(p, res.getString("twitchid"), res.getString("authtoken"));
 					} else {
 						p.sendMessage(ChatColor.RED + "No authentication details found");
-						p.spigot().sendMessage(this.authmsg);
```
Comment (Dkamps18) [inline src/main/java/net/dkamps18/channelpoints/Handler/CommandHandler.java:48]: You can't send a Text component with the normal api

File: src/main/java/net/dkamps18/channelpoints/Handler/CommandHandler.java [hunk 2]
```diff
@@ -84,7 +82,7 @@ public boolean run(CommandSender sender, Command cmd, String label, String[] arg
 						}
 					}
 				} else {
-					p.spigot().sendMessage(this.authmsg);
+					p.sendMessage(this.authmsg);
```
Comment (Dkamps18) [inline src/main/java/net/dkamps18/channelpoints/Handler/CommandHandler.java:85]: You can't send a Text component with the normal api

File: src/main/java/net/dkamps18/channelpoints/Handler/PubSubHandler.java [hunk 3]
```diff
@@ -47,11 +50,11 @@ public void onMessage(String message) {
 									Location loc = p.getLocation();
 									p.setVelocity(new Vector(loc.getDirection().getX() * 80, loc.getY() * 40, loc.getDirection().getZ() * 80));
 									PubSubHandler.this.plugin.disablefall.add(p.getUniqueId());
-									ResultSet dbr = PubSubHandler.this.plugin.dbu.get_user_data(p.getUniqueId().toString());
+									ResultSet dbr = PubSubHandler.this.plugin.databaseUtil.getUserData(p.getUniqueId().toString());
 									try {
 										if (dbr.next()) {
 											// See this awesome code by Marenthyu to make the patch requests proxy through https://github.com/Dkamps18/MC-ChannelPoints/issues/1
-											// PubSubHandler.this.plugin.tapi.acceptreward(rid, d.getAsJsonObject("redemption").get("id").getAsString(), dbr.getString("twitchid"), dbr.getString("authtoken"));
+											PubSubHandler.this.plugin.tapi.acceptreward(rid, d.getAsJsonObject("redemption").get("id").getAsString(), dbr.getString("twitchid"), dbr.getString("authtoken"));
```
Comment (Dkamps18) [inline src/main/java/net/dkamps18/channelpoints/Handler/PubSubHandler.java:57]: This was commented out due to it being for the event only shouldn't really be used was more a emergency fix and therefore commented out cause it won't be up anymore

**Review by Dkamps18 — CHANGES_REQUESTED** [review]

File: src/main/java/net/dkamps18/channelpoints/Handler/PubSubHandler.java (hunk 3 above)
Comment (DeJayDev) [reply src/main/java/net/dkamps18/channelpoints/Handler/PubSubHandler.java:57]: In the copy I saw it wasn't commented, so I just undid it.

File: src/main/java/net/dkamps18/channelpoints/Handler/CommandHandler.java (hunk 1 above)
Comment (DeJayDev) [reply src/main/java/net/dkamps18/channelpoints/Handler/CommandHandler.java:48]: Doable with PaperSpigot, I just assumed that was the usual but I can revert this.

File: src/main/java/net/dkamps18/channelpoints/Handler/CommandHandler.java (hunk 2 above)
Comment (DeJayDev) [reply src/main/java/net/dkamps18/channelpoints/Handler/CommandHandler.java:85]: Doable with PaperSpigot, I just assumed that was the usual but I can revert this.

**Pull request closed without merging**